core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -641,4 +591,165 @@ public OptionalInt getTraceSamplingPriority(final String authId) {
+            return result;
+        }
+    }
+
+    /**
+     * Get the set of subject distinguished names configure for this tenant.
+     * 
+     * @return The set of subject distinguished names or an empty set if no CA has been configured for this tenant.
+     */
+    @JsonIgnore
+    public Set<X500Principal> getTrustedCaSubjectDns() {
+
+        if (trustConfigurations == null) {
+            return Collections.emptySet();
+        } else {
+            final Set<X500Principal> subjectNames = new HashSet<>();
+            trustConfigurations.keySet().forEach(subjectDn -> {
+                subjectNames.add(new X500Principal(subjectDn));
+            });
+            return subjectNames;
+        }
+    }
+
+    /**
+     * Get the list of configured trusted CAs for this tenant.
+     * 
+     * @return The list of configured trusted CAs or {@code null} if this tenant has no configured trusted CA.
+     *
+     */
+    @JsonIgnore
+    public List<JsonObject> getTrustedCAs() {
+
+        if (trustConfigurations == null) {
+            return null;
+        } else {
+            return trustConfigurations.values().stream().flatMap(Collection::stream).collect(Collectors.toList());
+        }
+
+    }
+
+    @JsonIgnore
+    private TrustAnchor getTrustAnchor(final JsonObject trustedCa) throws GeneralSecurityException {
+
+        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
+        final String encodedKey = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, String.class);
+        final String type = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_KEY_ALGORITHM, String.class,
+                "RSA");
+
+        final Instant notBefore = getNotBefore(trustedCa);
+        final Instant notAfter = getNotAfter(trustedCa);
+        final Instant now = Instant.now();
+
+        if (subjectDn == null || encodedKey == null || notBefore == null || notAfter == null) {
+            return null;
+        } else if (now.isBefore(notBefore) || now.isAfter(notAfter)) {
+            // trusted CA is either too early for use or expired
+            return null;
+        } else {
+            final PublicKey pubKey = getPublicKey(encodedKey, type);
+            return new TrustAnchor(subjectDn, pubKey, null);
+        }
+    }
+
+    /**
+     * Add a trusted CA configuration for this tenant.
+     * 
+     * @param trustedCa The trust configuration to add.
+     * 
+     * @return This tenant so the API can be used fluently.
+     */
+    private TenantObject addTrustedCA(final JsonObject trustedCa) {
+
+        if (trustConfigurations == null) {
+            trustConfigurations = new HashMap<>();
+        }
+        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
+        if (Strings.isNullOrEmpty(subjectDn)) {
```

Comment:
this check should be at the start of the method, in order to prevent unnecessary creation of the HashMap